### PR TITLE
Asian

### PR DESCRIPTION
**Pull request by kevingivens**
Adding analytic asian pricers and instruments.  Feedback welcomed.  In particular, I wonder if it's better to group asian pricing engine in same file.

### CODE REVIEW

File: quantlib/test/test_asian.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+        tolerance = 1.0e-4
+
+        self.assertAlmostEqual(4.6922, option.net_present_value, delta=tolerance)
+        
+        # trying to approximate the continuous version with the discrete version
+        running_accumulator = 1.0
+        past_fixings = 0
+    
+        ##fixingDates(exerciseDate-today+1);
```
Comment (dpinte) [inline quantlib/test/test_asian.py:119]: any reason to keep the commented code?

**Review by dpinte — CHANGES_REQUESTED** [review]
@kevingivens PR looks good to me.  I've just left a note about commented code.

File: quantlib/test/test_asian.py (hunk 1 above)
Comment (kevingivens) [reply quantlib/test/test_asian.py]: I added more tests and removed this line in 062a6cd7e77997265e545249fa0131382b2bf7ce
The only issue I have now is a failing test for the theta greek.  I commented it out for now.

Comment (dpinte) [reply quantlib/test/test_asian.py]: :+1:

**Review by dpinte — APPROVED** [review]
LGTM

**Comment by kevingivens** [comment]
thanks a lot

**Pull request merged**